Storage API Changes (Part 2 - Simplify the Storage user experience)

tg-msft opened the pull request:
I'm adding this on top of the existing Storage API changes, so please only look at the last commit (see #6718 for Part 1):

- Rename `BlobClient` => `Specialized.BlobBaseClient`
- Add `BlobClient : BlobBaseClient` with an `Upload` method for block blobs
- Add overloads to ease parameter choice for `BlobBaseClient.Download`

tg-msft committed:
(Storage) API cleanup per Arch Board

- Move shared code into folders
- Remove unused constants

tg-msft committed:
Simple moves and renames

- Move HttpRange into Common
- Rename ID to Id
- Rename cancellation to cancellationToken
- Move access conditions into .Models
- Move remaining options into .Models
- Change Enqueue to return a single item instead of an array

tg-msft committed:
Move to Azure.Core's HttpRange

tg-msft committed:
Add simple Upload, Download, and BlobClient

kfarmer-msft commented on sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:
```diff
@@ -0,0 +1,2191 @@
+        /// </remarks>
+        /// <param name="versionId">The version ID to use on this blob. An empty string or null indicates to use the base blob.</param>
+        /// <returns>The new <see cref="BlobClient"/> instance referencing the verionId.</returns>
+        //public virtual BlobBaseClient WithVersionId(string versionId) => this.WithVersionIdImpl(versionId);
```
Status of the commented methods?

tg-msft replied:
These were commented out by Sean right before //build.

kfarmer-msft reviewed (approved):
Question about commented methods.

pakrym commented on sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs:
```diff
@@ -49,1522 +68,334 @@ public class BlobClient
+                metadata: metadata,
+                progressHandler: progressHandler,
+                cancellationToken: cancellationToken)
+                .EnsureCompleted();
```
This calls async method without passing isAsync

tg-msft replied:
Good catch - fixed.

tg-msft committed:
CR feedback and sync tests

tg-msft committed:
Fix sync APIs per Pavel's feedback

tg-msft committed:
Adding test recordings

    pushd $EnlistmentRoot
    $projects = "Common", "Blobs", "Queues", "Files"
    foreach ($project in $projects) {
        $source = ".\artifacts\bin\Azure.Storage.$($project).Tests\Debug\netcoreapp2.1\SessionRecords\"
        $dest = ".\sdk\storage\Azure.Storage.$($project)\tests\SessionRecords\"
        robocopy /xc /xn /xo /s $source $dest
    }
    popd

tg-msft committed:
Splitting apart Errors per CR feedback

tg-msft committed:
Test error message fix

pakrym commented on sdk/storage/Azure.Storage.Blobs/tests/SessionRecords/ServiceClientTests/SetPropertiesAsyncAsync.json:
```diff
@@ -0,0 +1,235 @@
+      "RequestBody": [
+        "\u003cStorageServiceProperties\u003e\r\n",
+        "  \u003cLogging\u003e\r\n",
+        "    \u003cVersion\u003e1.0\u003c/Version\u003e\r\n",
```
I've noticed that the request body is pretty-printed, is it worth disabling the pretty-printing to save bandwidth?

seanmcc-msft replied:
These are data for recorded tests, correct?  If so, I don't think we are actually using any bandwidth.

pakrym replied:
Recorded tests store exact thing that was sent over the wire during the Live test.

tg-msft replied:
Yes - filed #6806.
